### PR TITLE
Add docs for addressable_light display

### PR DESCRIPTION
**Pull request by justfalter**
## Description:

Adds support for addressable light displays

**Related issue (if applicable):** 

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1272

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/display/addressable_light.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+- **width** (**Required**, int): The width of the LED matrix in pixels.
+- **height** (**Required**, int): The height of the LED matrix in pixels.
+- **rotation** (*Optional*): Set the rotation of the display. Everything you draw in ``lambda:`` will be rotated
+  by this option. One of ``0°`` (default), ``90°``, ``180°``, ``270°``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to call the lambda to update the display.
+  Defaults to ``16ms``.
+- **pixel_mapper** (*Optional*, :ref:`lambda <config-lambda>`): A lambda that returns the integer address of the LED
+  given the supplied the ``x`` and ``y`` pixel coordinate. By default, a left-to-right direct pixel mapper is used. 
+- **lambda** (*Optional*, :ref:`lambda <config-lambda>`): The lambda to use for rendering the content on the display.
+  ``it`` will be an instance of :apiclass:`AddressableLightDisplay <addressable_light::AddressableLightDisplay>`.
```
Comment (jesserockz) [inline components/display/addressable_light.rst]: According to your python file `it` will be casted down to a `DisplayBuffer`

https://github.com/esphome/esphome/pull/1272/files#diff-9e44ad364e074c2728ed271806dc0305814ff1365a49b2e2c6fd3a9c6001ab36R59-R63

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/display/addressable_light.rst (hunk 1 above)
Comment (jesserockz) [reply components/display/addressable_light.rst]: Either way is fine, just making sure the docs match the code

Comment (justfalter) [reply components/display/addressable_light.rst]: Hmm.. Now I'm wondering if it's important that it be an `AddressableLightDisplay` ref during the render lambda... 
The public stuff hanging off of it that's not in DisplayBuffer are:
- get_light
- set_width
- set_height
- set_enabled
- get_enabled
- set_pixel_mapper
- display

I think it'd feel pretty weird to access any of those functions during the lambda, so I'll just update the documentation to reflect the implementation.

Comment (jesserockz) [reply components/display/addressable_light.rst]: Yeah, generally you don't need to access the specific public methods of the display platform in the lambda.

Comment (jesserockz) [reply components/display/addressable_light.rst]: You can always get them with `id(display_id).xxxx`

Comment (justfalter) [reply components/display/addressable_light.rst]: All done!

**Review by jesserockz — APPROVED** [review]

**Pull request merged**